devito/compiler.py

```diff
@@ -12,7 +12,18 @@
 
 
 class Compiler(GCCToolchain):
-    """Base class for all compiler classes."""
+    """Base class for all compiler classes.
+
+    :param openmp: Boolean indicating if openmp is enabled. False by default
+
+    The base class defaults all compiler specific settings to empty lists. Preset configurations
+    can be built by inheriting from `Compiler` and setting specific flags to the desired values, eg.:
+
+    def class MyCompiler(Compiler):
+        def __init__(self):
+            self.cc = 'mycompiler'
+            self.cflags = ['list', 'of', 'all', 'compiler', 'flags']
```

Comment:
I think this doc string should specify which settings can be changed by subclasses, this doesn't tell much.